app/scripts/migrations/105.test.ts

```diff
@@ -285,5 +296,41 @@ describe('migration #105', () => {
         new Error(`state.PreferencesController?.selectedAddress is undefined`),
       );
     });
+
+    it('recovers from invalid selectedAddress state', async () => {
+      const expectedUUID = addressToUUID(MOCK_ADDRESS);
+
+      const oldData = {
+        PreferencesController: {
+          identities: {
+            [MOCK_ADDRESS]: { name: 'Account 1', address: MOCK_ADDRESS },
+          },
+          selectedAddress: undefined,
+        },
+      };
+      const oldStorage = {
+        meta: { version: 103 },
+        data: oldData,
+      };
+
+      const newStorage = await migrate(oldStorage);
+
+      expect(newStorage.data).toStrictEqual({
+        PreferencesController: expect.objectContaining({
+          selectedAddress: MOCK_ADDRESS,
+        }),
+        AccountsController: {
+          internalAccounts: {
+            accounts: {
+              [expectedUUID]: expectedInternalAccount(
+                MOCK_ADDRESS,
+                `Account 1`,
```

Comment:
I would have expect the linter to complain about this one (backtick vs `'`) while there's no "variables to expand"